src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpMethodRepoDAO.scala

```diff
@@ -0,0 +1,32 @@
+package org.broadinstitute.dsde.rawls.dataaccess
+
+import akka.actor.ActorSystem
+import org.broadinstitute.dsde.rawls.model.AgoraEntity
+import org.broadinstitute.dsde.rawls.model.WorkspaceJsonSupport._
+import scala.concurrent.Await
+import scala.concurrent.duration.Duration
+import scala.util.Try
+import spray.client.pipelining._
+import spray.http.HttpCookie
+import spray.http.HttpHeaders.Cookie
+import spray.httpx.SprayJsonSupport._
+
+/**
+ * @author tsharpe
+ */
+class HttpMethodRepoDAO( methodRepoServiceURL: String )( implicit system: ActorSystem ) extends MethodRepoDAO {
+
+  override def getMethodConfig( namespace: String, name: String, version: String, authCookie: HttpCookie ): Option[AgoraEntity] = {
+    val url = s"${methodRepoServiceURL}/configurations/${namespace}/${name}/${version}"
+    import system.dispatcher
+    val pipeline = addHeader(Cookie(authCookie)) ~> sendReceive ~> unmarshal[AgoraEntity]
+    Try(Await.result(pipeline(Get(url)),Duration.Inf)).toOption
```

Comment:
First, how is that a monstrosity?

Second, you are creating an internal API, what is the expected outcome of getMethodConfig? As we discussed, it seems like there are 3 cases: some, none, exception. If you say we want to distinguish between different kind of exceptions then it is a different story but I thought we agreed that we were not.

For your precise definition, I think your monstrous code hits the nail on the head. Only 404 means none. If you get a successful response code, that means some. All else is an exception. Please pack as much relevant info into the exception message as possible so that we can give a good message to the caller.